src/common/endoflife.py

```diff
@@ -74,3 +75,37 @@ def write_releases(product, releases, pathname="releases") -> None:
             # sort by date then version (desc)
             sorted(releases.items(), key=lambda x: (x[1], x[0]), reverse=True)
         ), indent=2))
+
+
+def parse_date(text, formats=frozenset([
+    "%B %d, %Y",  # January 1, 2020
+    "%b %d, %Y",  # Jan 1, 2020
+    "%B %d %Y",   # January 1 2020
+    "%b %d %Y",   # Jan 1 2020
+    "%d %B %Y",   # 1 January 2020
+    "%d %b %Y",   # 1 Jan 2020
+    "%d-%b-%Y",   # 1-Jan-2020
+    "%d-%B-%Y",   # 1-January-2020
+    "%B %Y",      # January 2020
+    "%b %Y",      # Jan 2020
+])): return parse_datetime(text, False, formats)
+
+
+def parse_datetime(text, to_utc=True, formats=frozenset([
+    "%Y-%m-%d %H:%M:%S",         # 2023-05-01 08:32:34
+    "%Y-%m-%d %H:%M:%S %z",      # 2023-05-01 08:32:34 +0900
+    "%a, %d %b %Y %H:%M:%S %Z",  # Wed, 01 Jan 2020 00:00:00 GMT
+    "%Y-%m-%dT%H:%M:%S%z",       # 2023-05-01T08:32:34+0900
+])):
+    """Parse a date from text using a list of formats.
+    """
+    text = text.strip()
+    for formats in formats:
+        try:
+            date = datetime.strptime(text, formats)
```

Comment:
Avoid reusing `formats` and the builtin `format`:
```suggestion
    for format_ in formats:
        try:
            date = datetime.strptime(text, format_)
```